File: src/main/java/com/iuh/quy/service/RouteService.java
package com.iuh.quy.service;

import java.util.Date;
import java.util.List;

import com.iuh.quy.model.Routes;

public interface RouteService {
	Routes saveRoute(Routes route);
	List<Routes> getAllRoutes();
	Routes getRouteById(Long route_id);
	String deleteRoute(Long route_id);
	Routes editRoute(Routes route, Long route_id);
	List<Routes> searchRoutes(String city_from, String city_to);
}

File: src/service/route.service.js
import axios from "axios";

const API_URL = "http://localhost:8080";

class RouteService{
    saveRoute(route){
        return axios.post(API_URL + "/saveRoute", route);
    }

    getAllRoutes(){
        return axios.get(API_URL + "/getAllRoutes");
    }

    getRouteById(id){
        return axios.get(API_URL + "/getRouteById/" + id);
    }

    deleteRoute(id){
        return axios.get(API_URL + "/deleteRoute/" + id);
    }

    editRoute(route){
        return axios.post(API_URL + "/editRoute/" + route.id, route);
    }
}
export default new RouteService;